File: GAM to Admob/java/RewardCustomEvent.java


import android.app.Activity;
import android.content.Context;

import androidx.annotation.NonNull;

import com.google.android.gms.ads.AdError;
import com.google.android.gms.ads.AdListener;
import com.google.android.gms.ads.FullScreenContentCallback;
import com.google.android.gms.ads.LoadAdError;
import com.google.android.gms.ads.OnUserEarnedRewardListener;
import com.google.android.gms.ads.admanager.AdManagerAdRequest;
import com.google.android.gms.ads.mediation.MediationAdLoadCallback;
import com.google.android.gms.ads.mediation.MediationConfiguration;
import com.google.android.gms.ads.mediation.MediationRewardedAd;
import com.google.android.gms.ads.mediation.MediationRewardedAdCallback;
import com.google.android.gms.ads.mediation.MediationRewardedAdConfiguration;
import com.google.android.gms.ads.rewarded.RewardItem;
import com.google.android.gms.ads.rewarded.RewardedAd;
import com.google.android.gms.ads.rewarded.RewardedAdLoadCallback;

public class RewardCustomEvent extends AdListener implements MediationRewardedAd {
    private RewardedAd rewardedAd;

    private final MediationRewardedAdConfiguration mediationRewardedAdConfiguration;


    private final MediationAdLoadCallback<MediationRewardedAd, MediationRewardedAdCallback>
            mediationAdLoadCallback;
    private MediationRewardedAdCallback rewardedAdCallback;

    /** Constructor. */
    public RewardCustomEvent(
            @NonNull MediationRewardedAdConfiguration mediationRewardedAdConfiguration,
            @NonNull MediationAdLoadCallback<MediationRewardedAd, MediationRewardedAdCallback>
                    mediationAdLoadCallback) {
        this.mediationRewardedAdConfiguration = mediationRewardedAdConfiguration;
        this.mediationAdLoadCallback = mediationAdLoadCallback;
    }
    public void loadAd() {
        // All custom events have a server parameter named "parameter" that returns
        // back the parameter entered into the AdMob UI when defining the custom event.

        String serverParameter = mediationRewardedAdConfiguration
                .getServerParameters()
                .getString(MediationConfiguration
                        .CUSTOM_EVENT_SERVER_PARAMETER_FIELD);

        AdManagerAdRequest adRequest = new AdManagerAdRequest.Builder().build();
        RewardedAd.load(mediationRewardedAdConfiguration.getContext(), serverParameter,
                adRequest, new RewardedAdLoadCallback() {
                    @Override
                    public void onAdFailedToLoad(@NonNull LoadAdError loadAdError) {
                        rewardedAd = null;
                    }
                    @Override
                    public void onAdLoaded(@NonNull RewardedAd ad) {
                        rewardedAd = ad;
                    }
                });
    }
    @Override
    public void showAd(@NonNull Context context) {
        if (rewardedAd != null) {
            rewardedAd.show((Activity) mediationRewardedAdConfiguration.getContext(), new OnUserEarnedRewardListener() {
                @Override
                public void onUserEarnedReward(@NonNull RewardItem rewardItem) {
                }
            });
        }
    }
}

File: GAM to Admob/kotlin/RewardCustomEvent.kt


import android.app.Activity
import android.content.Context
import com.google.android.gms.ads.AdListener
import com.google.android.gms.ads.LoadAdError
import com.google.android.gms.ads.admanager.AdManagerAdRequest
import com.google.android.gms.ads.mediation.*
import com.google.android.gms.ads.rewarded.RewardedAd
import com.google.android.gms.ads.rewarded.RewardedAdLoadCallback

class RewardCustomEvent
/** Constructor.  */(
    private val mediationRewardedAdConfiguration: MediationRewardedAdConfiguration,
    private val mediationAdLoadCallback: MediationAdLoadCallback<MediationRewardedAd, MediationRewardedAdCallback>
) : AdListener(), MediationRewardedAd {
    private var rewardedAd: RewardedAd? = null
    private val rewardedAdCallback: MediationRewardedAdCallback? = null
    fun loadAd() {
        // All custom events have a server parameter named "parameter" that returns
        // back the parameter entered into the AdMob UI when defining the custom event.
        val serverParameter = mediationRewardedAdConfiguration
            .serverParameters
            .getString(MediationConfiguration.CUSTOM_EVENT_SERVER_PARAMETER_FIELD)
        val adRequest = AdManagerAdRequest.Builder().build()
        RewardedAd.load(
            mediationRewardedAdConfiguration.context, serverParameter!!,
            adRequest, object : RewardedAdLoadCallback() {
                override fun onAdFailedToLoad(loadAdError: LoadAdError) {
                    rewardedAd = null
                }

                override fun onAdLoaded(ad: RewardedAd) {
                    rewardedAd = ad
                }
            })
    }

    override fun showAd(context: Context) {
        if (rewardedAd != null) {
            rewardedAd!!.show((mediationRewardedAdConfiguration.context as Activity)) { }
        }
    }
}